### DISCUSSION TITLE
ILP and SDP calls in YALMIP

### DISCUSSION THREAD

**Discussion by ArjunM21** [yalmip/YALMIP#1266] [General discussion]
I wanted to know if it is possible to get the number of ILP or SDP calls made by a solver in YALMIP. I want this number to be outputted as the solver solves the problem.

**Comment by johanlofberg** [comment]
Unclear what context you are talking about, but https://yalmip.github.io/Extracting is the answer

**Reply by ArjunM21** [reply]
Thank you but I already have gone through this site.  This does not specify what exactly I want.

Please find an image of the kind of information I am looking for. Kindly tell me if there is something like this that fcan be extracted from the solver.

![image](https://user-images.githubusercontent.com/91719302/233978027-a17d6864-c09c-438c-8a85-62ff4ada57cf.png)

**Reply by johanlofberg** [reply]
You haven't even told us which solver you are talking about.

Regardless though, the last sentence on the page is the answer to the question

**Reply by ArjunM21** [reply]
I am using multiple solvers for different codes. I use sdpt3 as well as sedumi. In that image, he found out the SDP calls made by the solver that he used. Can that be done for sedumi or sdpt3?

**Reply by johanlofberg** [reply]
in that image some integer solver is used, so it makes no sense to compare with sdpt3 or sedumi as those aren't integer solvers. it is very unclear what you are asking or what you want to compare

**Reply by ArjunM21** [reply]
I don't want to compare anything. I want to know if it is possible to get the number of SDP calls made by sedumi or sdpt3. This is so that when I run multiple types of codes using sedumi or sdpt3, I can make a comparison of the number of SDP calls made among them. 

For example, if I run a code using sedumi with has a parameter 'a' and then run the same code using sedumi with parameter 'b', I want to make a comparison of the number of SDP calls made by sedumi. I'm not trying to compare to an integer solvers, but with multiple instances of sedumi itself.

**Reply by johanlofberg** [reply]
That question makes no sense. Sedumi is an SDP solver, so it does not make any calls to any SDP solver. If you call Sedumi, one call to an SDP solver will be made: the one you just made.

An integer solver based on branch&bound will solve SDPs in relaxation nodes, and thus call Sedumi or SDPT3 etc repeatedly, and then it makes sense to talk about how many calls the solver has made to an SDP solver

**Reply by ArjunM21** [reply]
Okay that made things clear. So in the same way will sedumi make ILP calls then?

**Reply by johanlofberg** [reply]
You seem to fail to understand what sedumi (or sdpt3) do, and how that is completely different from integer programming.

Sedumi will not make a single call to any external solver. It is a complete solver which solves linear continuous conic models, very different from integer programming.

**Reply by ArjunM21** [reply]
Okay everything makes sense now. Thank you so much for your help.
